Xamarin.Essentials/Screenshot/MediaFile.shared.cs

```diff
@@ -0,0 +1,43 @@
+﻿using System;
+using System.IO;
+
+namespace Xamarin.Essentials
+{
+    public class MediaFile : IDisposable
```

Comment:
We just merged in the file sharing bits, and as a result we got a new `FileBase` type. Could you see if you can re-use that type? We want to use that as a base type for all files. So, for this PR, you would maybe have a `ScreenshotFile : FileBase` and you can add any new methods you need to the new type. We hope that this will work, and you are the first to do something more than what we have. Let us know what doesn't work so we can make that base type be useful.

Have a look at how we did `EmailAttachment` and you can see what we are trying to do.

Reply:
Just took a look,we can definitely reuse the FileBase. I would love to see virtual methods for reading the file though, as on some platforms we can ellide the temporary file creation for the screenshot, which would be preferrable imho. then, if we were to use the same FileBase in another api where it would be passed in (e.g.) EmailAttachment, we could materialize the file in the file system as needed and not earlier. With this we could also add in-memory files to other api's easily.